modules/cache.py

```diff
@@ -83,6 +87,49 @@ def _downsize_cache_to_target_bytes(self, target_bytes:int):
     def clear(self):
         self._downsize_cache_to_target_bytes(0)
 
+    def populate_cache(self, cache_file: str):
+        try:   
+            # open the file and read the data
+            with open(cache_file, "r") as f:
+                
+                # json.load converts the json data into python dictionary
+                dict_data = json.load(f)
+            
+            # populate the cache
+            for _, video_info in dict_data.items():
+                self.add(video_info["url"])
```

Comment:
we shouldnt call `add`, because wouldnt that re-download the video? we have the video downloaded already.

we should be setting `self.video_id_to_path` directly in this file

Reply:
but the downloaded videos will be cleared when the server shuts down. The` self.video_id_to_path` only stores the metadata of the videos, but not the actual downloaded file. If we don't redownload the video, it's no use to store the video's metadata

so if we don't call the add function, then we either have to keep the downloaded videos somewhere in the file system or store the downloaded file in the cache file (not sure if it's possible). Is that what you want?

Reply:
the point of this feature is so we dont clear the videos, and write the cache state to a file if its specified